cli/slsa-verifier/verify/verify_artifact.go

```diff
@@ -40,39 +40,55 @@ type VerifyArtifactCommand struct {
 }
 
 func (c *VerifyArtifactCommand) Exec(ctx context.Context, artifacts []string) (*utils.TrustedBuilderID, error) {
-	artifactHash, err := getArtifactHash(artifacts[0])
-	if err != nil {
-		return nil, err
-	}
+	var builderId *utils.TrustedBuilderID
 
-	provenanceOpts := &options.ProvenanceOpts{
-		ExpectedSourceURI:      c.SourceURI,
-		ExpectedBranch:         c.SourceBranch,
-		ExpectedDigest:         artifactHash,
-		ExpectedVersionedTag:   c.SourceVersionTag,
-		ExpectedTag:            c.SourceTag,
-		ExpectedWorkflowInputs: c.BuildWorkflowInputs,
-	}
+	for _, artifact := range artifacts {
+		artifactHash, err := getArtifactHash(artifact)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Verifying artifact %s: FAILED: %v\n\n", artifact, err)
+			return nil, err
+		}
 
-	builderOpts := &options.BuilderOpts{
-		ExpectedID: c.BuilderID,
-	}
+		provenanceOpts := &options.ProvenanceOpts{
+			ExpectedSourceURI:      c.SourceURI,
+			ExpectedBranch:         c.SourceBranch,
+			ExpectedDigest:         artifactHash,
```

Comment:
apologies for only seeing this now! 

likely it probably makes the most sense to migate `ExpectedDigest` to `ExpectedDigests` so that we don't have to run signature verification more than once. I realize that's a breaking change -- maybe we should file a cleanup issue before the next major release?

doing it in a backwards compatible way with a new field would i guess be possible, and we could check for both ExpectedDigest/ExpectedDigests. It's not urgent to me though

Reply:
That's a good idea. Created #422 to track

Reply:
is it breaking for CLI or just the API? If just for the API, we could add a section to the README saying that we respect semver for the CLI only at the moment.

Reply:
Yeah, it would be just for the API -- the CLI would still have the versioning. 

Interesting about CLI semver -- in the past I've experienced only API being the versioning importance, but actually here CLI may be the only thing. No one is using as a library yet..

Reply:
I like what OSV does: the call out exactly what follows semver https://github.com/google/osv-scanner#semver-adherence. Created https://github.com/slsa-framework/slsa-github-generator/issues/1427 for tracking.